tests/rptest/tests/maintenance_test.py

```diff
@@ -0,0 +1,146 @@
+
+        self.logger.debug(
+            "Checking that node {node.name} is not in maintenance mode")
+        status = self.admin.maintenance_status(node)
+        assert status[
+            "draining"] == False, f"Node {node.name} in maintenance mode"
+
+        self.admin.maintenance_start(node)
+
+        self.logger.debug(
+            "Waiting for node {node.name} to enter maintenance mode")
+        wait_until(lambda: self._in_maintenance_mode(node),
+                   timeout_sec=30,
+                   backoff_sec=5)
+
+        self.logger.debug("Waiting for node {node.name} leadership to drain")
+        wait_until(lambda: not self._has_leadership_role(node),
+                   timeout_sec=60,
+                   backoff_sec=10)
+
+        self.logger.debug(
+            "Waiting for node {node.name} maintenance mode to complete")
+        wait_until(lambda: self._in_maintenance_mode_fully(node),
+                   timeout_sec=60,
+                   backoff_sec=10)
+
+    def _disable_maintenance(self, node):
+        wait_until(lambda: not self._in_maintenance_mode(node),
+                   timeout_sec=30,
+                   backoff_sec=5)
+
+        wait_until(lambda: self._has_leadership_role(node),
+                   timeout_sec=120,
+                   backoff_sec=10)
+
+    def _verify_cluster(self, target, target_expect):
+        for node in self.redpanda.nodes:
+            expect = False if node != target else target_expect
+            wait_until(
+                lambda: self._in_maintenance_mode(node) == expect,
+                timeout_sec=30,
+                backoff_sec=5,
+                err_msg=f"expected {node.name} maintenance mode: {expect}")
+
+    @cluster(num_nodes=3)
+    def test_maintenance(self):
+        target = random.choice(self.redpanda.nodes)
+        self._enable_maintenance(target)
+        self.admin.maintenance_stop(target)
+        self._disable_maintenance(target)
+
+    @cluster(num_nodes=3, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_maintenance_sticky(self):
+        nodes = random.sample(self.redpanda.nodes, len(self.redpanda.nodes))
+        for node in nodes:
+            self._enable_maintenance(node)
+            self._verify_cluster(node, True)
+
+            self.redpanda.restart_nodes(node)
+            self._verify_cluster(node, True)
+
+            self.admin.maintenance_stop(node)
+            self._disable_maintenance(node)
+            self._verify_cluster(node, False)
+
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+        self._verify_cluster(None, False)
+
+    @cluster(num_nodes=3)
+    def test_exclusive_maintenance(self):
+        target, other = random.sample(self.redpanda.nodes, k=2)
+        assert target is not other
+        self._enable_maintenance(target)
+        try:
+            self._enable_maintenance(other)
+        except requests.exceptions.HTTPError as e:
+            if "invalid state transition" in e.response.text:
```

Comment:
nit: could also assert the status code (it's nice to distinguish 500s and 400s especially)

Reply:
fixed